src/coord/src/catalog.rs

```diff
@@ -2616,6 +2660,20 @@ impl<S: Append> Catalog<S> {
                         self.state
                             .pack_compute_instance_replica_update(compute_id, &name, -1),
                     );
+
+                    let compute_instance = &self.state.compute_instances_by_id[&compute_id];
+                    let replica_id = compute_instance.replica_id_by_name[&name];
+                    let replica = &compute_instance.replicas_by_id[&replica_id];
+                    for process_id in replica.process_status.keys() {
+                        let update = self.state.pack_compute_instance_status_update(
+                            compute_id,
+                            replica_id,
+                            *process_id,
+                            -1,
+                        );
+                        builtin_table_updates.push(update);
```

Comment:
It's more conventional to ensure the order of operations drop dependent items before their dependencies to ensure e.g. no foreign key violations, so this should like just go to https://github.com/MaterializeInc/materialize/pull/12853/files#diff-6a43b2224b24c06a172d1d39239372d062bc7f231b770021901afc8c5c60998eR2659

Reply:
The link doesn't work, but I assume you mean the status updates should be pushed before the replica update.

Doesn't that mean the order is also wrong for `Op::DropComputeInstance` above, where the instance update is pushed before the introspection index updates?

Reply:
Yep!